periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
- keep https://github.com/kubeflow/model-registry/pull/153
- keep https://github.com/kubeflow/model-registry/pull/158
- keep https://github.com/kubeflow/model-registry/pull/155
- keep https://github.com/kubeflow/model-registry/pull/156
- keep https://github.com/kubeflow/model-registry/pull/157
- keep https://github.com/kubeflow/model-registry/pull/151
- keep https://github.com/kubeflow/model-registry/pull/148
- keep https://github.com/kubeflow/model-registry/pull/137
- resolve merge conflicts 66800203c1b2dc343fcc30b192e7aa964a048cf3
(`.gitignore` diverged, now aligned)

tarilabs commented on internal/core/core.go:
```diff
@@ -191,10 +199,20 @@ func (serv *modelRegistryService) GetRegisteredModels(listOptions ListOptions) (
+		log.Printf("Updating model version %s", *modelVersion.Id)
+	}
+
+	if parentResourceId == nil {
+		return nil, fmt.Errorf("missing registered model id, cannot create model version without registered model")
+	}
+
+	registeredModel, err := serv.GetRegisteredModelById(*parentResourceId)
```
will followup for this check to go into 204 and for the parentResourceId to be derived (making it truly optional in method signature)

tarilabs replied:
👉 https://github.com/opendatahub-io/model-registry/issues/105

tarilabs reviewed (approved):
as discussed, thank you so much 🚀 💪

dhirajsb reviewed (approved):
apart from glog logging, lgtm

lampajr commented on internal/core/core.go:
```diff
@@ -96,27 +99,32 @@ func (serv *modelRegistryService) UpsertRegisteredModel(registeredModel *openapi
-func (serv *modelRegistryService) GetRegisteredModelById(id *BaseResourceId) (*openapi.RegisteredModel, error) {
-	log.Printf("Getting registered model %d", *id)
+func (serv *modelRegistryService) GetRegisteredModelById(id string) (*openapi.RegisteredModel, error) {
+	log.Printf("Getting registered model %s", id)
```
Thanks for the input, opened https://github.com/opendatahub-io/model-registry/issues/110 to keep track the change

tarilabs committed:
py(chore): bump to next version (#153)

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

dependabot[bot] committed:
build(deps): bump github.com/go-chi/chi/v5 from 5.0.12 to 5.0.14 (#158)

Bumps [github.com/go-chi/chi/v5](https://github.com/go-chi/chi) from 5.0.12 to 5.0.14.
- [Release notes](https://github.com/go-chi/chi/releases)
- [Changelog](https://github.com/go-chi/chi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-chi/chi/compare/v5.0.12...v5.0.14)

---
updated-dependencies:
- dependency-name: github.com/go-chi/chi/v5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump coverage from 7.5.3 to 7.5.4 in /clients/python (#155)

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.5.3 to 7.5.4.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.5.3...7.5.4)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump testcontainers from 4.5.1 to 4.6.0 in /clients/python (#156)

Bumps [testcontainers](https://github.com/testcontainers/testcontainers-python) from 4.5.1 to 4.6.0.
- [Release notes](https://github.com/testcontainers/testcontainers-python/releases)
- [Changelog](https://github.com/testcontainers/testcontainers-python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testcontainers/testcontainers-python/compare/testcontainers-v4.5.1...testcontainers-v4.6.0)

---
updated-dependencies:
- dependency-name: testcontainers
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump ruff from 0.4.9 to 0.4.10 in /clients/python (#157)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.4.9 to 0.4.10.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/v0.4.9...v0.4.10)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

antisaling committed:
REST: Fix PATCH method (#151)

* update goverter to 1.4.1

As that fixed a nil assignment override[1][2] which prevented us from
emulating copy constructors[3].
The update also allows us to use generics.

[1]: https://github.com/jmattheis/goverter/issues/146#issuecomment-2161457915
[2]: https://github.com/jmattheis/goverter/issues/97
[3]: https://github.com/jmattheis/goverter/issues/147

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>
Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

* simplify converter utils using generics

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

* server: update existing objects on PATCH

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

---------

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>
Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

ederign committed:
feat: client ui frontend scaffolding (#148)

In this PR:
- UI frontend scaffolding
- Github Action

Most of the content of this PR is extract from https://github.com/patternfly/patternfly-react-seed/tree/v6. Thank you so much patternfly team for the seed!

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

antisaling committed:
Add readthedocs config (#137)

* py: pyproject: create docs group

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

* py: add readthedocs config

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

---------

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

tarilabs committed:
periodic sync upstream KF to midstream ODH

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
